devito/ir/clusters/algorithms.py

```diff
@@ -543,10 +548,13 @@ def normalize_reductions_sparse(cluster, sregistry, options):
             # into
             # `s = f(u[t, i], src, ...)`
             # `u[t, i] += s`
-            name = sregistry.make_name()
-            v = Symbol(name=name, dtype=e.dtype)
-            processed.extend([e.func(v, e.rhs, operation=None),
-                              e.func(e.lhs, v)])
+            if e.rhs in mapper:
```

Comment:
note that just like normalize_reductions_dense this is flaky if for whatever reason one Eq writes to `e.rhs` in between these clusters, as it would violate a data dependence